mp4parse/src/lib.rs

```diff
@@ -1503,6 +1508,178 @@ fn read_infe<T: Read>(src: &mut BMFFBox<T>) -> Result<ItemInfoEntry> {
+    Channels(TryVec<u8>),
+    AuxiliaryType(TryString),
+    Unsupported,
+}
+
+impl ImageProperty {
+    fn clone(&self) -> Result<Self> {
+        Ok(match self {
+            Self::Channels(val) => Self::Channels(val.try_clone()?),
+            Self::AuxiliaryType(val) => Self::AuxiliaryType(val.try_clone()?),
+            Self::Unsupported => Self::Unsupported,
+        })
+    }
+}
+
+struct Association {
+    item_id: u32,
+    property_index: u16,
+}
+
+pub struct AssociatedProperty {
+    pub item_id: u32,
+    pub property: ImageProperty,
+}
+
+fn read_ipma<T: Read>(src: &mut BMFFBox<T>) -> Result<TryVec<Association>> {
+    let (version, flags) = read_fullbox_extra(src)?;
+
+    let mut associations = TryVec::new();
+
+    let entry_count = be_u32(src)?;
+    for _ in 0..entry_count {
+        let item_id = if version == 0 {
+            be_u16(src)? as u32
+        } else {
+            be_u32(src)?
+        };
+        let association_count = src.read_u8()?;
+        for _ in 0..association_count {
+            let first_byte = src.read_u8()?;
+            let essential_flag = first_byte & 1 << 7;
+            let value = first_byte - essential_flag;
+            let property_index = if flags & 1 != 0 {
+                ((value as u16) << 8) | src.read_u8()? as u16
+            } else {
+                value as u16
+            };
+            associations.push(Association {
+                item_id,
+                property_index,
+            })?;
+        }
+    }
+    Ok(associations)
+}
+
+fn read_ipco<T: Read>(src: &mut BMFFBox<T>) -> Result<TryVec<ImageProperty>> {
+    let mut properties = TryVec::new();
+
+    let mut iter = src.box_iter();
+    while let Some(mut b) = iter.next_box()? {
+        // Must push for every property to have correct index for them
+        properties.push(match b.head.name {
+            BoxType::PixelInformationBox => ImageProperty::Channels(read_pixi(&mut b)?),
+            BoxType::AuxiliaryTypeProperty => ImageProperty::AuxiliaryType(read_auxc(&mut b)?),
+            _ => {
+                skip_box_remain(&mut b)?;
+                ImageProperty::Unsupported
+            }
+        })?;
+    }
+    Ok(properties)
+}
+
+fn read_pixi<T: Read>(src: &mut BMFFBox<T>) -> Result<TryVec<u8>> {
+    let version = read_fullbox_version_no_flags(src)?;
+    if version != 0 {
+        return Err(Error::Unsupported("pixi version"));
+    }
+
+    let mut channels = TryVec::new();
+    let channel_count = src.read_u8()?;
+    for _ in 0..channel_count {
+        channels.push(src.read_u8()?)?;
+    }
+    Ok(channels)
+}
+
+fn read_auxc<T: Read>(src: &mut BMFFBox<T>) -> Result<TryString> {
+    let version = read_fullbox_version_no_flags(src)?;
+    if version != 0 {
+        return Err(Error::Unsupported("auxC version"));
+    }
+
+    let mut aux = TryString::new();
+    loop {
+        match src.read_u8()? {
+            0 => break,
+            c => aux.push(c)?,
+        }
+    }
+    Ok(aux)
```

Comment:
Technically the `auxC` box can contain additional bytes after the end of the nul-terminated `aux_type` field. My guess is we're unlikely to care about those values for now, but I think we should make sure to handle them correctly. Here's what I got trying to implement the spec fully:
```suggestion
    src.try_read_to_end(&mut aux)?;

    let (aux_type, aux_subtype): (TryString, TryVec<u8>);
    if let Some(nul_byte_pos) = aux.iter().position(|&b| b == b'\0') {
        let (a, b) = aux.as_slice().split_at(nul_byte_pos);
        aux_type = a.try_into()?;
        aux_subtype = (&b[1..]).try_into()?;
    } else {
        aux_type = aux;
        aux_subtype = TryVec::new();
    }

    Ok(AuxiliaryTypeProperty {
        aux_type,
        aux_subtype,
    })
```
This requires a new type:
```rust
#[derive(Debug, PartialEq)]
pub struct AuxiliaryTypeProperty {
    aux_type: TryString,
    aux_subtype: TryVec<u8>,
}
```

I'm also open to solutions which ignore the `aux_subtype` since we're not using it for anything as long as they still correctly parse `auxC` instances that contain it and avoid small reads.